staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator/aggregator.go

```diff
@@ -331,6 +409,16 @@ func (s *specAggregator) AddUpdateAPIService(handler http.Handler, apiService *v
 func (s *specAggregator) RemoveAPIServiceSpec(apiServiceName string) error {
 	s.rwMutex.Lock()
 	defer s.rwMutex.Unlock()
+	if utilfeature.DefaultFeatureGate.Enabled(features.OpenAPIV3) {
+		s.openAPIV3rwMutex.Lock()
+		if _, ok := s.openAPIV3Specs[apiServiceName]; ok {
+			for group := range s.openAPIV3Specs[apiServiceName] {
+				s.openAPIV3VersionedService.DeleteGroupVersion(group)
+			}
+			delete(s.openAPIV3Specs, apiServiceName)
+		}
+		s.openAPIV3rwMutex.Unlock()
```

Comment:
this is dangerous. Put the unlock into a defer. Otherwise, we risk a deadlock when `DeleteGroupVersion` panics.